### ISSUE TITLE
GHCR registry not working

### ISSUE THREAD

**Issue by blackstardlb** [fmartinou/whats-up-docker#219]
18:16:59.690 DEBUG whats-up-docker/watcher.docker.remote: 401 - {"errors":[{"code":"UNAUTHORIZED","message":"unauthenticated: User cannot be authenticated with the token provided."}]} (container=container)
    StatusCodeError: 401 - {"errors":[{"code":"UNAUTHORIZED","message":"unauthenticated: User cannot be authenticated with the token provided."}]}
        at new StatusCodeError (/home/node/app/node_modules/request-promise-core/lib/errors.js:32:15)
        at Request.plumbing.callback (/home/node/app/node_modules/request-promise-core/lib/plumbing.js:104:33)
        at Request.RP$callback [as _callback] (/home/node/app/node_modules/request-promise-core/lib/plumbing.js:46:31)
        at Request.self.callback (/home/node/app/node_modules/request/request.js:185:22)
        at Request.emit (node:events:513:28)
        at Request.emit (node:domain:489:12)
        at Request.<anonymous> (/home/node/app/node_modules/request/request.js:1154:10)
        at Request.emit (node:events:513:28)
        at Request.emit (node:domain:489:12)
        at IncomingMessage.<anonymous> (/home/node/app/node_modules/request/request.js:1076:12)
        at Object.onceWrapper (node:events:627:28)
        at IncomingMessage.emit (node:events:525:35)
        at IncomingMessage.emit (node:domain:489:12)
        at endReadableNT (node:internal/streams/readable:1358:12)
        at processTicksAndRejections (node:internal/process/task_queues:83:21)

My token works with 
`TOKEN=$(curl -u <username>:<personal_access_token> https://ghcr.io/token\?scope\="repository:<repository/path>:pull" | jq -r .token)`

`curl -H "Authorization: Bearer $TOKEN" https://ghcr.io/v2/<repository/path>/tags/list`

If i `echo $TOKEN | base64 -d` i simply get my personal_access_token back.
From what I can tell the base64 encoded token from this line https://github.com/fmartinou/whats-up-docker/blob/c5033ff3ce02fa157ebcfd5dec56607728a94651/app/registries/providers/ghcr/Ghcr.js#L56 should not include the login name

**Comment by fmartinou** [comment]
Hi,

I'm surprised because I have automatic tests in CI checking that the `GHCR` integration is working. :thinking:

I've just tested it locally with success.

What I did:
1. Created a new personal access token as described in the documentation
2. Configured wud with `WUD_REGISTRY_GHCR_USERNAME=<my email address>` and `WUD_REGISTRY_GHCR_TOKEN=ghp_2Qqb5X9ux8..................`

Then I started a `radarr` container pulled from ghcr
```yaml
version: '3'

services:
  radarr:
    image: ghcr.io/linuxserver/radarr:4.3.2
```

And finally, I started wud which has properly found a new version available on ghcr
```
15:45:36.969 DEBUG whats-up-docker/watcher.docker.local: Start watching (container=local_radarr)
15:45:36.971 DEBUG whats-up-docker/registry.ghcr: ghcr - Get linuxserver/radarr tags
15:45:39.745 DEBUG whats-up-docker/watcher.docker.local: Container watched for the first time (container=local_radarr)
15:45:39.754 DEBUG whats-up-docker/trigger.mock.example: Run trigger (container=local_radarr)
15:45:39.756  INFO whats-up-docker/trigger.mock.example:
    MOCK triggered title = 
    New tag found for container radarr
15:45:39.757  INFO whats-up-docker/trigger.mock.example:
    MOCK triggered body  = 
    Container radarr running with tag 4.3.2 can be updated to tag arm64v8-version-v0.2.0.1504
    
15:45:39.758  INFO whats-up-docker/watcher.docker.local: Cron finished (1 containers watched, 0 errors, 1 available updates)
```

Are you sure about the configuration you set on `wud`?

**Comment by blackstardlb** [comment]
Seems like it's an issue with private images

```
17:42:51.481  INFO whats-up-docker: What's up Docker? is starting (version = latest)
17:42:51.516  INFO whats-up-docker/store: Load store from (/store/wud.json)
17:42:51.528  INFO whats-up-docker/store: Create Collection app
17:42:51.534  INFO whats-up-docker/store: Migrate data from version undefined to version latest
17:42:51.541  INFO whats-up-docker/store: Create Collection containers
17:42:51.542  INFO whats-up-docker/prometheus: Init Prometheus module
17:42:51.594 DEBUG whats-up-docker: Start container metrics interval
17:42:53.140  INFO whats-up-docker/registry.ghcr: Register with configuration {"username":"brisondl@gmail.com","token":"g**************************************V"}
17:42:53.836  INFO whats-up-docker/registry.ecr: Register with configuration {}
17:42:53.848  INFO whats-up-docker/registry.hotio: Register with configuration {}
17:42:53.862  INFO whats-up-docker/registry.hub: Register with configuration {}
17:42:53.873  INFO whats-up-docker/registry.quay: Register with configuration {}
17:42:59.933  INFO whats-up-docker/watcher.docker.local: Register with configuration {"socket":"/var/run/docker.sock","watchbydefault":false,"port":2375,"cron":"0 * * * *","watchall":false,"watchevents":true}
17:42:59.954  INFO whats-up-docker/watcher.docker.local: Cron scheduled (0 * * * *)
17:43:00.137  INFO whats-up-docker/registry: No authentication configured => Allow anonymous access
17:43:00.183  INFO whats-up-docker/authentication.anonymous.anonymous: Register with configuration {}
17:43:00.186 DEBUG whats-up-docker/api: API/UI enabled => Start Http listener on port 3000
17:43:00.309  WARN whats-up-docker: Anonymous authentication is enabled; please make sure that the app is not exposed to unsecure networks
17:43:00.667  INFO whats-up-docker/api: Server listening on port 3000 (HTTP)
17:43:01.220  INFO whats-up-docker/watcher.docker.local: Cron started (0 * * * *)
17:43:01.545  INFO whats-up-docker/watcher.docker.local: Listening to docker events
17:43:03.492 DEBUG whats-up-docker/watcher.docker.local: Start watching (container=local_my_public_image)
17:43:03.512 DEBUG whats-up-docker/registry.ghcr: ghcr - Get blackstardlb/my-public-image/my-public-image tags
17:43:03.525 DEBUG whats-up-docker/watcher.docker.local: Start watching (container=local_my_private_image)
17:43:03.541 DEBUG whats-up-docker/registry.ghcr: ghcr - Get blackstardlb/my_private_image tags
17:43:03.552 DEBUG whats-up-docker/watcher.docker.local: Start watching (container=local_radar)
17:43:03.562 DEBUG whats-up-docker/registry.ghcr: ghcr - Get linuxserver/radarr tags
17:43:08.199 DEBUG whats-up-docker/watcher.docker.local: Container watched for the first time (container=local_my_public_image)
17:43:08.232  WARN whats-up-docker/watcher.docker.local: Error when processing (401 - {"errors":[{"code":"UNAUTHORIZED","message":"unauthenticated: User cannot be authenticated with the token provided."}]}) (container=local_my_private_image)
17:43:08.240 DEBUG whats-up-docker/watcher.docker.local: 401 - {"errors":[{"code":"UNAUTHORIZED","message":"unauthenticated: User cannot be authenticated with the token provided."}]} (container=local_my_private_image)
    StatusCodeError: 401 - {"errors":[{"code":"UNAUTHORIZED","message":"unauthenticated: User cannot be authenticated with the token provided."}]}
        at new StatusCodeError (/home/node/app/node_modules/request-promise-core/lib/errors.js:32:15)
        at Request.plumbing.callback (/home/node/app/node_modules/request-promise-core/lib/plumbing.js:104:33)
        at Request.RP$callback [as _callback] (/home/node/app/node_modules/request-promise-core/lib/plumbing.js:46:31)
        at Request.self.callback (/home/node/app/node_modules/request/request.js:185:22)
        at Request.emit (node:events:513:28)
        at Request.emit (node:domain:489:12)
        at Request.<anonymous> (/home/node/app/node_modules/request/request.js:1154:10)
        at Request.emit (node:events:513:28)
        at Request.emit (node:domain:489:12)
        at IncomingMessage.<anonymous> (/home/node/app/node_modules/request/request.js:1076:12)
        at Object.onceWrapper (node:events:627:28)
        at IncomingMessage.emit (node:events:525:35)
        at IncomingMessage.emit (node:domain:489:12)
        at endReadableNT (node:internal/streams/readable:1358:12)
        at processTicksAndRejections (node:internal/process/task_queues:83:21)
17:43:08.249 DEBUG whats-up-docker/watcher.docker.local: Container watched for the first time (container=local_my_private_image)
17:43:40.960 DEBUG whats-up-docker/watcher.docker.local: Container watched for the first time (container=local_radar)
17:43:40.965  INFO whats-up-docker/watcher.docker.local: Cron finished (3 containers watched, 1 errors, 1 available updates)
17:43:50.998  INFO whats-up-docker/watcher.docker.local: Status changed from running to restarting (container=local_my_public_image)
17:43:52.185  INFO whats-up-docker/watcher.docker.local: Status changed from restarting to running (container=local_my_public_image)
17:44:14.110  INFO whats-up-docker/watcher.docker.local: Status changed from running to restarting (container=local_my_public_image)
17:44:15.359  INFO whats-up-docker/watcher.docker.local: Status changed from restarting to running (container=local_my_public_image)
```

**Comment by blackstardlb** [comment]
With my base64 encoded token (with no username/email) i can reach https://ghcr.io/v2/linuxserver/radarr/tags/list and also the tag list for my private image.

**Comment by fmartinou** [comment]
I'm gonna fix; thank you for the troubleshoot 👍

**Comment by fmartinou** [comment]
[The fix](https://github.com/fmartinou/whats-up-docker/compare/master...fix/%23219_ghcr_token) is available in `fmartinou/whats-up-docker:fix__219_ghcr_token` and will be part of the next release version.

**Comment by blackstardlb** [comment]
Thanks for the fix 👍🏽

**Comment by fmartinou** [comment]
Released as `6.1.0`
